### PR TITLE
[GenericPoseGraph] Add IMU Calibration and Pose3D node types.

### PR DESCRIPTION
**Pull request by pifon2a**

### CODE REVIEW

**Comment by pifon2a** [comment]
@wally-the-cartographer merge

File: cartographer/pose_graph/node/imu_calibration.cc [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  // TODO(pifon): Use a common method to convert from Eigen::Quaterniond to
+  // proto. Probably, the one defined in transform.h.
+  auto* orientation = imu_calibration->mutable_orientation();
+  orientation->set_w(orientation_[0]);
+  orientation->set_x(orientation_[1]);
+  orientation->set_y(orientation_[2]);
+  orientation->set_z(orientation_[3]);
+
+  std::cout << parameters.DebugString();
```
Comment (mbokeloh) [inline cartographer/pose_graph/node/imu_calibration.cc]: Do you want that to be in there?

File: cartographer/pose_graph/node/nodes.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+#include "cartographer/pose_graph/node/pose_2d.h"
+#include "cartographer/pose_graph/node/pose_3d.h"
+
+#include <map>
+
+namespace cartographer {
+namespace pose_graph {
+
+struct Nodes {
+  // TODO(pifon): Should it really be an std::map or smth else?
```
Comment (mbokeloh) [inline cartographer/pose_graph/node/nodes.h]: Let's first try it with map but do an experiment once everything is done with an std::unordered_map to see if there is a difference.

File: cartographer/pose_graph/node/imu_calibration.cc [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+namespace cartographer {
+namespace pose_graph {
+
+ImuCalibration::ImuCalibration(const NodeId& node_id, bool constant,
+                               double gravity_constant,
+                               const Eigen::Quaterniond& orientation)
+    : Node(node_id, constant),
+      gravity_constant_(gravity_constant),
+      orientation_{{orientation.w(), orientation.x(), orientation.y(),
```
Comment (mbokeloh) [inline cartographer/pose_graph/node/imu_calibration.cc]: I assume that the order is important but it comes at a surprise. Could you add a comment somewhere why that is? Alternatively, could we keep the Eigen::Quaterniond format?

Also the same below in Pose3D. Maybe we can have a converter function?

File: cartographer/pose_graph/node/nodes.h (hunk 2 above)
Comment (pifon2a) [reply cartographer/pose_graph/node/nodes.h]: Agreed.

File: cartographer/pose_graph/node/imu_calibration.cc (hunk 3 above)
Comment (pifon2a) [reply cartographer/pose_graph/node/imu_calibration.cc]: I just kept Cartographer's format. But I agree, let's keep Eigen's.

**Comment by pifon2a** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user pifon2a. Merge queue now has a length of 1.

**Pull request merged**